clang/test/CIR/Lowering/nested-union-array.c

```diff
@@ -1,28 +0,0 @@
-// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-llvm %s -o %t.ll
-// RUN: FileCheck --input-file=%t.ll %s --check-prefix=LLVM
-
-struct nested
-{
-  union {
-    const char *single;
-    const char *const *multi;
-  } output;
-};
-static const char * const test[] = {
-  "test",
-};
-const struct nested data[] = 
-{
-    {
-        {
-            .multi = test,
-        },
-    },
-    {
-        {
-            .single = "hello",
-        },
-    },
-};
-
-// LLVM: @data = constant [2 x {{.*}}]
```

Comment:
Rather than remove this test completely, can you just update the check to show what will be generated after your change?

I'm seeing this:
```
@data = constant { { { ptr } }, { { ptr } } } { { { ptr } } { { ptr } { ptr @test } }, { { ptr } } { { ptr } { ptr @.str } } }, align 16
````
which looks quite reasonable.

I think the only reason classic codegen produced an array in this case is that `pointer to pointer to char` and `pointer to char` end up as the same type in LLVM IR. If you go back to clang 14 (i.e. before opaque pointers), you get structure-based initialization:

https://godbolt.org/z/qM4vz19b5